planner/core/logical_plan_builder.go

```diff
@@ -1321,6 +1324,109 @@ func (b *PlanBuilder) buildProjection(ctx context.Context, p LogicalPlan, fields
 		}
 	}
 	proj.SetChildren(p)
+	// delay the only-full-group-by-check in create view statement to later query.
+	if !b.isCreateView && b.ctx.GetSessionVars().OptimizerEnableNewOnlyFullGroupByCheck {
+		fds := proj.ExtractFD()
```

Comment:
I think this `ExtractFD()` should be put inside the `if` block below.

Reply:
Changed.